### PR TITLE
PLANNER-521- add optaplanner annotations domain editor functionality in kie distributions, but masked out by role plannermgmt

### PR DESCRIPTION
**Pull request by wmedvede**
- Planner domain added to kie-wb, and kie-drools-wb

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello Pere, would you mind review this PR?

It basically includes the adding of the Optaplanner Dock to the Authoring perspectives for both kie-wb and kie-drools-wb in a server side ApplicationScoped component.

**Comment by pefernan** [comment]
+1

File: kie-drools-wb/kie-drools-wb-webapp/src/main/resources/workbench-policy.properties [hunk 1]
```diff
@@ -45,6 +45,9 @@ feature.wb_release_project=Release Process
 # data modeller features
 feature.wb_data_modeler_edit_sources=Edit Java Sources
 
+# opta planner features
+feature.wb_optaplanner_domain=Opta Planner Domain
```
Comment (psiroky) [inline kie-drools-wb/kie-drools-wb-webapp/src/main/resources/workbench-policy.properties]: There should not be a space between Opta and Planner.

File: kie-wb/kie-wb-webapp/src/main/resources/workbench-policy.properties [hunk 2]
```diff
@@ -49,6 +49,9 @@ feature.wb_release_project=Release Process
 # data modeller features
 feature.wb_data_modeler_edit_sources=Edit Java Sources
 
+# opta planner features
+feature.wb_optaplanner_domain=Opta Planner Domain
```
Comment (psiroky) [inline kie-wb/kie-wb-webapp/src/main/resources/workbench-policy.properties]: Same as above. Better to keep the project name consistent.

**Comment by manstis** [comment]
Is kie-wb-distributions exempt from having to write tests? ;-)

**Comment by wmedvede** [comment]
@manstis Mike, this is part of a Perspective configuration and additionally nobody knows what the future of the Docks will be. In my opinion (for this particular case) to try to invent a test for this case just for full filling the bureaucracy  is a waste of time.

**Comment by psiroky** [comment]
This is not directly related to this PR, but will mention it anyway. We really should find a way to share the common stuff between kie-wb-webapp and kie-drools-wb-webapp config/setup code. Currently there is lot of duplication which sometimes results in things being updated only in one place. I know there are things that are only meant for kie-wb and it should be possible to cope with that (e.g. extending the base shared setup/config.)

**Comment by ge0ffrey** [comment]
@psiroky about that DRY violation issue: +1. There's a belief that the new plugin architecture will fix this (and it probably will), but I'd argue it's worth refactoring already now to share as much as possible. This will minimize the work that needs to be done once we migrate to that new plugin architecture.

**Comment by wmedvede** [comment]
@csadilek 

A test was added, so from my side it's ready to merge and I'll not waste more time on this.

**Comment by Rikkola** [comment]
@psiroky @ge0ffrey I think everything needed is already pluggable. Menu's, styles, perspectives and screens. The problem is not just kie-wb and kie-drools-wb. 

Every workbench has the same redundancy for styles, services and so on. A lot could be done. The copy&paste is just legacy from the days when we didn't have time to do these.

Kind of related to the "should we test experimental". Both kie-drools-wb and kie-wb still have this experimental setup that was there just to make them demoable ;)

**Comment by wmedvede** [comment]
@csadilek 
The "1 failing check" above is not related to the stuff pushed in this commit, this are the failing tests:

org.kie.smoke.wb.rest.KieRemoteRestSmokeIntegrationTest.testRestUrlsGroupAssignmentProcess
org.kie.smoke.wb.rest.KieRemoteRestSmokeIntegrationTest.testRestHistoryLogs
org.kie.smoke.wb.rest.KieRemoteRestSmokeIntegrationTest.testRestRemoteApiExtraJaxbClasses
org.kie.smoke.wb.rest.KieRemoteRestSmokeIntegrationTest.testRestRemoteApiRuleTaskProcess
org.kie.smoke.wb.rest.KieRemoteRestSmokeIntegrationTest.testRestRemoteApiHumanTaskProcess

**Comment by csadilek** [comment]
@wmedvede would you mind investigating why these tests fail? It doesn't matter if it's this commit's fault or not. We're one team and these tests are important. We all have to take ownership of these tests. Once you know the root cause, I am sure whoever caused the problem is eager to help.

In contrast to all other tests, we don't want to run the full integration test suite with every commit (it would take too long), so these type of tests will sometimes fail after a merge has happened. Whoever runs into the problem first (you are right now) should investigate (which shouldn't take long).

**Comment by wmedvede** [comment]
@csadilek 

The test seems to run locally and I've tried to find what could be related to other projects but I have not idea of what could be happening. In case it doesn't take long to you to find a clue, it will be very welcome.

**Comment by psiroky** [comment]
I've re-triggered the build on different slave to see if the failures are related to specific slave or not.

**Comment by wmedvede** [comment]
kieremotetests

It seems like the tests worked this time. I've chatted with @mrietveld  this evening and he commented that some weird cases of tests that stopped/started working in the jenkins jobs were also detected for some jobs. But don't know if there are a repeatable use case to check with. Btw, let's see what happens.

**Comment by csadilek** [comment]
6.4.x: 2b65ba7428ab2e1e385d784d601a9d62011dccf7

**Pull request closed without merging**